ipn/ipnlocal/profiles.go

```diff
@@ -66,7 +65,13 @@ func (pm *profileManager) SetCurrentUserID(uid ipn.WindowsUserID) error {
 	// the selected profile for the current user.
 	b, err := pm.store.ReadState(ipn.CurrentProfileKey(string(uid)))
 	if err == ipn.ErrStateNotExist || len(b) == 0 {
-		pm.NewProfile()
+		if runtime.GOOS == "windows" {
+			if err := pm.migrateFromLegacyPrefs(); err != nil {
```

Comment:
I think you need to handle `errAlreadyMigrated`?

Reply:
@maisem I guess the question is, what can we actually do to handle it at this point?

By reaching this point, we've already established that the current profile key is not yielding any profile info, and yet the migration code is telling us that we've already migrated. Given the inconsistency of that state, I'm not sure what else we can do other than propagate that message up the stack. Thoughts?

Reply:
good point, i guess it should error out